evm_loader/program/src/entrypoint.rs

```diff
@@ -388,72 +388,36 @@ fn process_instruction<'a>(
-                }
-            }
+            check_secp256k1_instruction(sysvar_info, unsigned_msg.len(), 9u16)?;
 
             let caller = H160::from_slice(from_addr);
             let trx: UnsignedTransaction = rlp::decode(unsigned_msg).map_err(|_| ProgramError::InvalidInstructionData)?;
 
             do_partial_call(program_id, step_count, accounts, trx.call_data, Some( (caller, trx.nonce) ))
         },
-        EvmInstruction::Continue {step_count} => { 
-            do_continue(program_id, step_count, accounts)
-        },
-        EvmInstruction::FinalizeCallFromRawEthereumTX { from_addr, sign, unsigned_msg } => {
+        EvmInstruction::Continue {} => {
             let account_info_iter = &mut accounts.iter();
             let _storage_info = next_account_info(account_info_iter)?;
             let _program_info = next_account_info(account_info_iter)?;
             let _caller_info = next_account_info(account_info_iter)?;
             let sysvar_info = next_account_info(account_info_iter)?;
 
-            let current_instruction = instructions::load_current_index(&sysvar_info.try_borrow_data()?);
-            let index = current_instruction - 1;
+            let (step_count, from_addr, sign, unsigned_msg) = {
+                let (_tag, rest) = instruction_data.split_first().ok_or(ProgramError::InvalidInstructionData)?;
+                let (step_count, rest) = rest.split_at(8);
+                let step_count = step_count.try_into().ok().map(u64::from_le_bytes).ok_or(ProgramError::InvalidInstructionData)?;
+                let (from_addr, rest) = rest.split_at(20);
+                let (sign, unsigned_msg) = rest.split_at(65);
 
-            match load_instruction_at(index.try_into().unwrap(), &sysvar_info.try_borrow_data()?) {
-                Ok(instr) => {
-                    if instr.program_id == secp256k1_program::id() {
-                        let reference_instruction = make_secp256k1_instruction(current_instruction, unsigned_msg.len(), 1u16);
-                        if reference_instruction != instr.data {
-                            debug_print!("wrong keccak instruction data");
-                            debug_print!(&("instruction: ".to_owned() + &hex::encode(&instr.data)));    
-                            debug_print!(&("reference: ".to_owned() + &hex::encode(&reference_instruction)));    
-                            return Err(ProgramError::InvalidInstructionData);
-                        }
-                    } else {
-                        return Err(ProgramError::IncorrectProgramId);
-                    }
-                },
-                Err(err) => {
-                    debug_print!("ERR");
-                    return Err(ProgramError::MissingRequiredSignature);
-                }
-            }
+                check_secp256k1_instruction(sysvar_info, unsigned_msg.len(), 9u16)?;
+
+                (step_count, from_addr, sign, unsigned_msg)
```

Comment:
Why do you parse instruction data here? It would be better to parse this instruction like `PartialCallFromRawEthereumTX`.

Reply:
When we continue execution from the account data, there will be no instruction.

Reply:
`step_count` should be passed through instruction_data. Another info will be get from account data.